### PR TITLE
kube-prometheus: Add Prometheus 2.0 rules

### PR DESCRIPTION
**Pull request by brancz**
@ant31

### CODE REVIEW

File: contrib/kube-prometheus/assets/prometheus/rules/prometheus.rules.yaml [hunk 1]
```diff
@@ -42,3 +42,30 @@ groups:
...
+    for: 12h
+    labels:
+      severity: warning
+    annotations:
+      description: '{{$labels.job}} at {{$labels.instance}} had {{$value | humanize}}
+        reload failures over the last four hours.'
+      summary: Prometheus has issues reloading data blocks from disk
+  - alert: PrometheusTSDBCompactionsFailing
+    expr: increase(prometheus_tsdb_compactions_failed_total[2h]) > 0
+    for: 12h
```
Comment (ant31) [inline contrib/kube-prometheus/assets/prometheus/rules/prometheus.rules.yaml]: ?

Comment (brancz) [reply contrib/kube-prometheus/assets/prometheus/rules/prometheus.rules.yaml]: this means that there have to be compaction fails within two hour periods for the past 12 hours

File: contrib/kube-prometheus/assets/prometheus/rules/prometheus.rules.yaml [hunk 2]
```diff
@@ -42,3 +42,30 @@ groups:
...
+    labels:
+      severity: warning
+    annotations:
+      description: '{{$labels.job}} at {{$labels.instance}} had {{$value | humanize}}
+        compaction failures over the last four hours.'
```
Comment (Quentin-M) [inline contrib/kube-prometheus/assets/prometheus/rules/prometheus.rules.yaml]: Where does the `four hours` come from?

Comment (brancz) [reply contrib/kube-prometheus/assets/prometheus/rules/prometheus.rules.yaml]: whoops, you're right, seems like that was mistaken with the alerting rule below

**Pull request merged**